### PR TITLE
[openal-soft] Remove removing macro in dynamic triplets

### PR DESCRIPTION
**Pull request by BillyONeal**
Add workaround suggested by Robert in #18098

### CODE REVIEW

**Review by PhoebeHui — APPROVED** [review]

**Comment by BillyONeal** [comment]
Dropping reviewed: this is ready to land as far as I'm concerned but I would like @DanielGibson and/or @HarrievG to confirm that their bug is actually fixed before merging.

**Comment by HarrievG** [comment]
checking dhewm3's dedicated build with VCPKG/CMAKE right now.

File: ports/openal-soft/portfile.cmake [hunk 1]
```diff
@@ -80,7 +80,10 @@ foreach(HEADER al.h alc.h)
     if(VCPKG_LIBRARY_LINKAGE STREQUAL "static")
         string(REPLACE "defined(AL_LIBTYPE_STATIC)" "1" AL_H "${AL_H}")
     else()
```
Comment (HarrievG) [inline ports/openal-soft/portfile.cmake:82]: you want to keep the empty if case?

**Comment by HarrievG** [comment]
Fix Confirmed .
[install-x86-windows-rel-out.log](https://github.com/microsoft/vcpkg/files/7227866/install-x86-windows-rel-out.log)
[install-x64-windows-dbg-out.log](https://github.com/microsoft/vcpkg/files/7227867/install-x64-windows-dbg-out.log)
[install-x64-windows-rel-out.log](https://github.com/microsoft/vcpkg/files/7227868/install-x64-windows-rel-out.log)
[install-x86-windows-dbg-out.log](https://github.com/microsoft/vcpkg/files/7227869/install-x86-windows-dbg-out.log)

File: ports/openal-soft/portfile.cmake (hunk 1 above)
Comment (DanielGibson) [reply ports/openal-soft/portfile.cmake:82]: probably to put the comment in?

**Comment by BillyONeal** [comment]
As the issue reporters reported that this fixes it, landed.

File: ports/openal-soft/portfile.cmake (hunk 1 above)
Comment (BillyONeal) [reply ports/openal-soft/portfile.cmake:82]: Yes, the empty else() is to indicate that we didn't forget and it is intentional.

**Pull request merged**